src/event_scanner/modes/historic.rs

```diff
@@ -0,0 +1,208 @@
+use alloy::{
+    eips::BlockNumberOrTag,
+    network::Network,
+    providers::RootProvider,
+    transports::{TransportResult, http::reqwest::Url},
+};
+
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+
+use crate::{
+    block_range_scanner::{BlockRangeScanner, ConnectedBlockRangeScanner, MAX_BUFFERED_MESSAGES},
+    event_scanner::{
+        EventScannerError,
+        filter::EventFilter,
+        listener::EventListener,
+        message::EventScannerMessage,
+        modes::common::{ConsumerMode, handle_stream},
+    },
+};
+
+pub struct HistoricScannerBuilder {
+    block_range_scanner: BlockRangeScanner,
+    // Defaults to Earliest
+    from_block: BlockNumberOrTag,
+    // Defaults to Latest
+    to_block: BlockNumberOrTag,
+}
+
+pub struct HistoricEventScanner<N: Network> {
+    config: HistoricScannerBuilder,
+    block_range_scanner: ConnectedBlockRangeScanner<N>,
+    listeners: Vec<EventListener>,
+}
+
+impl HistoricScannerBuilder {
+    pub(super) fn new() -> Self {
```

Comment:
```suggestion
    #[must_use]
    pub(super) fn new() -> Self {
```
applies to other builders' `new`